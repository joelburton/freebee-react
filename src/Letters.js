import Letter from "./Letter";
import "./Letters.css";
import {useState} from "react";
import {shuffle} from "lodash";

function Letters({letters, center}) {
  const [randomLetters, setRandomLetters] = useState(Array.from(letters));

  function randomize() {
    setRandomLetters(letters => shuffle(letters));
  }

  return (
      <div className="Letters">
        <Letter letter={center} isCenter />
        { randomLetters.map(letter => (
            <Letter key={letter} letter={letter}/>
        ))}
        <button className="Letters-randomize" onClick={randomize}>âŸ²</button>
      </div>
  );
}

export default Letters;