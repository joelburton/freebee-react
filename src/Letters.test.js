import lodash from "lodash";

lodash.shuffle = jest.fn();

// NOPE NOPE NOPE
// import react from "react";
// react.useState =

import {fireEvent, render} from "@testing-library/react";
import Letters from "./Letters";

it("renders without crashing", function () {
  render(<Letters letters="ABC" center="Z"/>);
});

it("shows letter", function () {
  const {getByText} = render(<Letters letters="ABC" center="Z"/>);

  const a = getByText("A");
  const b = getByText("B");
  const c = getByText("C");
  const z = getByText("Z");
  expect(a).toBeInTheDocument();
  expect(b).toBeInTheDocument();
  expect(c).toBeInTheDocument();
  expect(z).toBeInTheDocument();
});

it("randomizes", function () {
  lodash.shuffle.mockReturnValueOnce(["A", "B", "C", "D", "F", "E"]);
  const {container } = render(<Letters letters="ABCDEF" center="Z"/>);
  // container=<div><Letters /></div>
  expect(container.querySelector(".Letter:nth-child(6)")).toHaveTextContent("E");
  expect(container.querySelector(".Letter:nth-child(7)")).toHaveTextContent("F");
  fireEvent.click(container.querySelector(".Letters-randomize"));
  expect(container.querySelector(".Letter:nth-child(6)")).toHaveTextContent("F");
  expect(container.querySelector(".Letter:nth-child(7)")).toHaveTextContent("E");
})

it("matches snapshot", function () {
  const {container} = render(<Letters letters="ABC" center="Z"/>);
  expect(container).toMatchSnapshot();
});
