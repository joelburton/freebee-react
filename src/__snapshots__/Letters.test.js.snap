// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Letters"
  >
    <div
      class="Letter Letter-isCenter"
    >
      <span>
        Z
      </span>
    </div>
    <div
      class="Letter "
    >
      <span>
        A
      </span>
    </div>
    <div
      class="Letter "
    >
      <span>
        B
      </span>
    </div>
    <div
      class="Letter "
    >
      <span>
        C
      </span>
    </div>
    <button
      class="Letters-randomize"
    >
      ‚ü≤
    </button>
  </div>
</div>
`;
